source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -240,15 +240,16 @@ bool Filter::parseExtensions(int fd) {
     // buf_ is no longer in use so we re-use it to read/discard
     int bytes_avail;
     auto& os_syscalls = Api::OsSysCallsSingleton::get();
-    if (os_syscalls.ioctl(fd, FIONREAD, &bytes_avail) < 0) {
+    if (os_syscalls.ioctl(fd, FIONREAD, &bytes_avail).rc_ < 0) {
       throw EnvoyException("failed to read proxy protocol (no bytes avail)");
     }
     if (bytes_avail == 0) {
       return false;
     }
     bytes_avail = std::min(size_t(bytes_avail), sizeof(buf_));
     bytes_avail = std::min(size_t(bytes_avail), proxy_protocol_header_.value().extensions_length_);
-    ssize_t nread = os_syscalls.recv(fd, buf_, bytes_avail, 0);
+    const Api::SysCallSizeResult result = os_syscalls.recv(fd, buf_, bytes_avail, 0);
+    ssize_t nread = result.rc_;
```

Comment:
Do we nee the ssize_t local var here? Same elsewhere in this function? Is a similar simplification available any other places that I didn't notice?

Reply:
Thought it'd keep the code more readable. I can simplify it here and at other places if required.

Reply:
I would probably drop the local variables where possible, but I don't feel that strongly about it. (I think the code would stay readable as long as the return value variable name is readable).